pr_testing/test_multiple_prs.sh

```diff
@@ -44,6 +44,37 @@ function get_pr_relval_args() {
   echo "${WF_ARGS}"
 }
 
+# Function to extract filenames by headername and append to indirectly-changed-files
+function extract_filenames() {
+  local headername="$1"
+  local input_file="./etc/dependencies/usedby.out"
+  local output_file="$WORKSPACE/indirectly-changed-files"
+  
+  # Extract lines starting with headername, split them, and append each filename to the temp file
+  grep "^$headername" "$input_file" | while read -r line; do
+    # Split the line into an array
+    IFS=' ' read -r -a array <<< "$line"
+
+    # Loop through each element after the first (which is the headername)
+    for filename in "${array[@]:1}"; do
+      echo "$filename" >> "$output_file"
+    done
+  done
+}
+
+# Function to get indirectly changed files
+function process_changed_files() {
+  [ -e "$WORKSPACE/indirectly-changed-files" ] || touch "$WORKSPACE/indirectly-changed-files"
+  # Iterate over each line in $WORKSPACE/changed-files
+  while IFS= read -r headername; do
+    # Call the function to extract filenames and append them atomically
+    extract_filenames "$headername"
+  done < "$WORKSPACE/changed-files"
+  local temp_file=$(mktemp)
+  sort -u $WORKSPACE/changed-files $WORKSPACE/indirectly-changed-files > $temp_file
+  mv $temp_file $WORKSPACE/changed-files
```

Comment:
@iarspider , we are overriding the original `$WORKSPACE/changed-files` file. I thought we discussed that we should create a separate file for direct + indirect change files

Reply:
we discussed that as `changed-files` is also used for other purposes so it is safer to use a different file for warnings. So please create a separate file. Better to name it such that it end up with `.txt` extension so that https://github.com/cms-sw/cms-bot/blob/master/pr_testing/_helper_functions.sh#L82 can upload it ( just for debugging purposes)

Reply:
> You suggested to pre-fill changed-files with directly changed files, then update it after running gcc compilation
 
correct, it should have both direct and indirect file but it should be in different file than `changed-files`